### PR TITLE
Refactor company edit views

### PR DESCRIPTION
**Pull request by teneightfive**
This change reduces the duplicated views used to edit different types of companies.

The differences are relatively minimal and have been handled using `if/else` blocks in one shared view for now.

### Future changes (out of scope for this PR)

- Switch to new form building method and new form macros
- Reduce duplication in controllers and middleware

### CODE REVIEW

File: src/apps/companies/controllers/ltd.js [hunk 1]
```diff
@@ -66,11 +50,15 @@ async function addDetails (req, res, next) {
     } else {
       res.locals.formData = companyFormService.getDefaultLtdFormForCH(res.locals.chCompany)
     }
+    res.locals.chDetailsLabels = chDetailsLabels
```
Comment (ztolley) [inline src/apps/companies/controllers/ltd.js]: move into own middleware as this is used in multiple plaves

File: src/apps/companies/middleware/form.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  next()
+}
+
+async function handleFormPost (req, res, next) {
+  try {
+    const savedCompany = await companyFormService.saveCompanyForm(req.session.token, req.body)
+
+    req.flash('success', 'Company record updated')
+    res.redirect(`/companies/view/foreign/${savedCompany.id}`)
```
Comment (ztolley) [inline src/apps/companies/middleware/form.js]: carries over the old bug to redirect to foreign. After this, we merge the company details view too

Comment (teneightfive) [reply src/apps/companies/middleware/form.js]: Yes, was aware of this. Hadn't finished with this method yet. Was planning to fix it with a follow up commit but it was getting late last night...

File: src/apps/companies/controllers/ltd.js (hunk 1 above)
Comment (teneightfive) [reply src/apps/companies/controllers/ltd.js]: Yeah, haven't quite nailed all of this yet and was kind of a temporary measure until I found a better place for it.

Comment (teneightfive) [reply src/apps/companies/controllers/ltd.js]: Actually, I'm not sure what you mean here. Do you mean the labels are used in multiple views or that this middleware is used in multiple other places?

File: src/apps/companies/views/edit.njk [hunk 3]
```diff
@@ -0,0 +1,147 @@
+{% extends "./_layout-edit.njk" %}
+
+{% block main_grid_right_column %}
+  {{ super() }}
```
Comment (ztolley) [inline src/apps/companies/views/edit.njk]: Do we need super?

Comment (teneightfive) [reply src/apps/companies/views/edit.njk]: Nope. Good spot.

**Pull request merged**